ykutil/src/obj.rs

```diff
@@ -21,11 +22,7 @@ pub static SELF_BIN_PATH: LazyLock<PathBuf> = LazyLock::new(|| {
     if addr == ptr::null_mut() as *mut c_void {
         panic!("couldn't find address of main()");
     }
-    let mut info = MaybeUninit::<Dl_info>::uninit();
-    if unsafe { dladdr(addr, info.as_mut_ptr()) } == 0 {
-        panic!("couldn't find Dl_info for main()");
-    }
-    let info = unsafe { info.assume_init() };
+    let info = dladdr(addr as usize).unwrap(); // ptr to usize cast always safe.
```

Comment:
I think this can be `usize::from(addr)`?

Also, is the `unwrap` guaranteed not to fail?

Reply:
I generally don't find `expect` super helpful, so I'm OK with the `unwrap` here, but I like to document it with something like "// If this fails, there's nothing useful we can do other than exit".

Reply:
Fixed in e5a024335eae9846fc47681829db0e5e0ed15587

Reply:
What about the `usize::from` comment?

Reply:
Woops, missed that.

Whats the advantage actually? We always cast from the same abi to the same abi.

I guess if some code changed around it, we would catch mistakes?

Reply:
Well, we can't use `from` without an additional `as` here anyway:
```
the trait bound `usize: From<*const c_void>` is not satisfied
```

Reply:
If we can't use it, fair enough.